### ISSUE TITLE
Workaround go multipart parser issue -- use different boundary?

### ISSUE THREAD

**Issue by tleyden** [couchbase/couchbase-lite-ios#1059]
Spun out of https://github.com/couchbase/couchbase-lite-ios/issues/1053

It looks like [certain boundary strings](https://github.com/golang/go/issues/14085#issuecomment-174347244) in `multipart/related` requests are triggering issue https://github.com/golang/go/issues/14085 with the go multipart parser.

If I change [TestReadmultipartbody4](https://gist.github.com/tleyden/c50334e3f3750911403d#file-testreadmultipartbody4) to remove the leading `-`, it no longer reproduces the issue.

What about changing couchbase-lite to use "safer" boundary strings to work around this?  I don't know what safer means, other than avoiding leading `-`'s (though I don't know for sure if that's the only class of boundary string that has issues)

**Comment by snej** [comment]
Can do. The change was unintentional; when I changed the format of document UUIDs I didn't consider that the same function was used to generate separators.

**Comment by snej** [comment]
OK, fixed on the master branch. Boundaries are again in the traditional UUID style, e.g. `C592FDB5-3E27-475B-92F5-1354BC813130`. Guaranteed no leading/trailing dashes. FYI the dashes were introduced  in 4/2015 by commit 611dca4a which changed the prefix in the strings generated by CBLCreateUUID() from a "!" to a "-". Which I think means this issue has been around since 1.1.

We should consider taking this fix for 1.2, since it's very low-risk — it simply changes one call to `CBLCreateUUID()` into a call to `NSUUID`. What think you, @zgramana ?

**Comment by tleyden** [comment]
I just verified the fix in https://github.com/couchbase/couchbase-lite-ios/commit/c7f4d221ddcf9b8332f959aa949efc245b313413 with the deepstyle sample app -- no longer seeing the same error.

I did notice some unexpected 503 errors from Sync Gateway, but it seems that the client successfully retried and pushed the documents.

Also, I think the last time I tested I used cblite 1.1 (sqlite), and it successfully upgraded the db to 1.2/master (sqlite).  [Relevant client logs](https://gist.github.com/tleyden/ad5f51e55518ea67a0da#file-client-logs-L1-L2)

[Client + SG logs](https://gist.github.com/tleyden/ad5f51e55518ea67a0da)
